### PR TITLE
Some updates to make lib more friendly to newcomers

### PR DESCRIPTION
**Pull request by mrsombre**
- Added support of `--help` option
- Added examples and links into README
- Changed main.go a bit for easy start
- Added go.mod support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
+## Quick start
+
+### Prerequisites
+
+- Install golang >= 1.11  
+    https://golang.org/dl/
+- Install Starcraft II game  
+    https://starcraft2.com/en-us/ (free version is okay)
+- Download Ladder  
+    https://drive.google.com/file/d/18lmZEzzZEP1VhqmHKsiSn9g7P2BDutNJ/view
```
Comment (chippydip) [inline README.md]: Might be better to just point to the channel topic in the #ladder channel in Discord so this link doesn't get out of date?

**Review by chippydip — COMMENTED** [review]
1) Is the NilBot even useful now that we have better examples?

2) I personally only use `go build` when testing with the ladder manager, for testing directly I always just use `go run`, is that worth pointing out to new users do you think?

File: README.md (hunk 1 above)
Comment (mrsombre) [reply README.md]: Yeah, added info about !maps command

**Comment by mrsombre** [comment]
- Dropped main.go in favor of examples
- Added `stub_bot` example
- Changed readme a bit to reflect changes

**Comment by mrsombre** [comment]
`go run` intended to be used w/ only single file, not good for many files, better to use go build in such case.

**Review by chippydip — APPROVED** [review]
Looks good! 👍

**Pull request merged**